common/network_manager.go

```diff
@@ -34,6 +35,15 @@ type Credentials struct {
 	Password string `js:"password"`
 }
 
+// IsEmpty returns true if the credentials are empty.
+func (c Credentials) IsEmpty() bool {
+	c = Credentials{
+		Username: strings.TrimSpace(c.Username),
+		Password: strings.TrimSpace(c.Password),
+	}
+	return c == (Credentials{})
```

Comment:
Preferably the `TrimSpace` would have been done when the object was created rather than when `IsEmpty` was called. It leaves us with a scenario where it might be possible to accidentally work with credentials that have fields with spaces.

Could we not have a validator/cleanup method that is called to remove the spaces after parsing/exporting these from the option?

Reply:
`IsEmpty` ensures that the `Credentials` value is truly empty (with none of its fields set), so I've tried to preserve the previous behavior. Still, it makes sense to add validator methods to the options, including this one 👍

Reply:
I've added an issue here: https://github.com/grafana/xk6-browser/issues/1530